src/argilla_sdk/records/_resource.py

```diff
@@ -182,87 +182,7 @@ def from_dict(
-                schema_item = schema.get(attribute)
-                if len(attribute_mapping) > 1:
-                    attribute_type = attribute_mapping[1]
-                if len(attribute_mapping) > 2:
-                    sub_attribute = attribute_mapping[2]
-            elif schema_item is mapping is None and attribute != "id":
-                warnings.warn(
-                    message=f"""Record attribute {attribute} is not in the schema so skipping. 
-                        Define a mapping to map source data fields to Argilla Fields, Questions, and ids
-                        """
-                )
-                continue
-
-            if attribute == "id":
-                record_id = value
-                continue
-
-            # Add suggestion values to the suggestions
-            if attribute_type == "suggestion":
-                if sub_attribute in ["score", "agent"]:
-                    suggestion_values[attribute][sub_attribute] = value
-
-                elif sub_attribute is None:
-                    suggestion_values[attribute].update(
-                        {"value": value, "question_name": attribute, "question_id": schema_item.id}
-                    )
-                else:
-                    warnings.warn(
-                        message=f"Record attribute {sub_attribute} is not a valid suggestion sub_attribute so skipping."
-                    )
-                continue
-
-            # Assign the value to question, field, or response based on schema item
-            if isinstance(schema_item, TextField):
-                fields[attribute] = value
-            elif isinstance(schema_item, QuestionType) and attribute_type == "response":
-                responses.append(Response(question_name=attribute, value=value, user_id=user_id))
-            elif isinstance(schema_item, QuestionType) and attribute_type is None:
-                suggestion_values[attribute].update(
-                    {"value": value, "question_name": attribute, "question_id": schema_item.id}
-                )
-            elif isinstance(schema_item, VectorField):
-                vectors.append(Vector(name=attribute, values=value))
-            elif isinstance(schema_item, MetadataType):
-                metadata[attribute] = value
-            else:
-                warnings.warn(message=f"""Record attribute {attribute} is not in the schema or mapping so skipping.""")
-                continue
-
-        suggestions = [Suggestion(**suggestion_dict) for suggestion_dict in suggestion_values.values()]
-
-        return cls(
-            id=record_id,
-            fields=fields,
-            suggestions=suggestions,
-            responses=responses,
-            vectors=vectors,
-            metadata=metadata,
-            external_id=data.get("external_id") or record_id,
-            _dataset=dataset,
-        )
+        return _ingest_record_from_dict(dataset=dataset, data=data, mapping=mapping, user_id=user_id)
```

Comment:
If the .from_dict just calls this internal function I would say to keep the code here and use directly Record.from_dict as is.